include/lsst/afw/detection/Footprint.h

```diff
@@ -340,212 +235,19 @@ class Footprint : public lsst::daf::base::Citizen,
-                      lsst::afw::image::Image<std::uint16_t>::Ptr dist);
-
-/**
-   Merges two Footprints -- appends their peaks, and unions their
-   spans, returning a new Footprint.
-
-   This const version requires that both input footprints are
-   normalized (and will raise an exception if not).
- */
-PTR(Footprint) mergeFootprints(Footprint const& foot1, Footprint const& foot2);
-
-/**
-   Merges two Footprints -- appends their peaks, and unions their
-   spans, returning a new Footprint.
- */
-PTR(Footprint) mergeFootprints(Footprint& foot1, Footprint& foot2);
-
-/**
- * Shrink a footprint isotropically by nGrow pixels, returning a new Footprint.
- */
-PTR(Footprint) shrinkFootprint(Footprint const& foot, int nGrow, bool isotropic);
-
-/**
- * Grow a Footprint by nGrow pixels, returning a new Footprint.
- */
-PTR(Footprint) growFootprint(Footprint const& foot, int nGrow, bool isotropic=true);
-
-/**
- * \note Deprecated interface; use the Footprint const& version.
- */
-PTR(Footprint) growFootprint(PTR(Footprint) const& foot, int nGrow, bool isotropic=true);
-
-/**
- * \brief Grow a Footprint in at least one of the cardinal directions,
- * returning a new Footprint
- *
- * Note that any left/right grow is done prior to the up/down grow, so
- * any left/right grown pixels \em are subject to a further up/down
- * grow (i.e. an initial single pixel Footprint will end up as a
- * square, not a cross.
- */
-PTR(Footprint) growFootprint(Footprint const& foot, int nGrow,
-                             bool left, bool right, bool up, bool down);
-
-/**
- * Return a list of BBox%s, whose union contains exactly the pixels in
- * foot, neither more nor less
- *
- * Useful in generating sets of meas::algorithms::Defects for the ISR
- */
-std::vector<lsst::afw::geom::Box2I> footprintToBBoxList(Footprint const& foot);
-
-/**
- * \brief Set all image pixels in a Footprint to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprint(ImageT *image,
-                                             Footprint const& footprint,
-                                             typename ImageT::Pixel const value);
-
-/**
- * \brief Set all image pixels in a set of Footprint%s to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprintList(ImageT *image,
-                                                 CONST_PTR(std::vector<PTR(Footprint)>) footprints,
-                                                 typename ImageT::Pixel  const value);
-
-/**
- * \brief Set all image pixels in a set of Footprint%s to a given value
- *
- * \return value
- */
-template<typename ImageT>
-typename ImageT::Pixel setImageFromFootprintList(ImageT *image,
-                                                 std::vector<PTR(Footprint)> const& footprints,
-                                                 typename ImageT::Pixel  const value);
```

Comment:
I suspect this is a fairly important convenience method that we want to preserve somehow, though it might be best reimplemented in pure Python as loop over an arbitrary iterable of `Footprints` (which would let it be used on a e.g. a generator that extracts `Footprint`s from an iterator over `SourceRecord`s).